src/conf_mode/load-balancing-inbound-ipvsadm.py

```diff
@@ -0,0 +1,152 @@
+from vyos import airbag
+airbag.enable()
+
+lvs_config = '/tmp/ipvsadm.rules'
+
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+    
+    base = ['load-balancing', 'inbound', 'ipvsadm']
+    conf = conf.get_config_dict(base, key_mangling=('-', '_'), get_first_key=True)
+    
+    if 'rule' in conf:
+        default_values = defaults(base + ['rule'])
+        for rule in conf['rule']:
+            conf['rule'][rule] = dict_merge(default_values, conf['rule'][rule])
+    
+    return conf
+
+def verify(config):
+    if not config:
+        return None
+    
+    if 'rule' in config:
+        ip_type = None
+        virtual_server_list = []
+        vip_list = []
+        for rule, rule_config in config['rule'].items():
+            if 'disable' in rule_config:
+                continue
+            
+            err_msg = f'Error in the configuration of rule {rule}. '
+            if 'virtual_ip' not in rule_config or 'virtual_port' not in rule_config:
+                raise ConfigError(err_msg + 'The virtual server IP and port must exist!')
+            
+            virtual_ip = rule_config['virtual_ip']
+            virtual_port = rule_config['virtual_port']
+            virtual_server = None
+            if is_ipv4(virtual_ip):
+                virtual_server = f'{virtual_ip}:{virtual_port}'
+            elif is_ipv6(virtual_ip):
+                virtual_server = f'[{virtual_ip}]:{virtual_port}'
+            
+            if virtual_server not in virtual_server_list:
+                virtual_server_list.append(virtual_server)
+            else:
+                raise ConfigError(err_msg + 'Do not allow duplicate virtual servers!')
+            
+            if virtual_ip not in vip_list:
+                vip_list.append(virtual_ip)
+        
+        for rule, rule_config in config['rule'].items():
+            if 'disable' in rule_config:
+                continue
+            
+            err_msg = f'Error in the configuration of rule {rule}. '
+            
+            if is_ipv4(rule_config['virtual_ip']):
+                ip_type = 4
+            elif is_ipv6(rule_config['virtual_ip']):
+                ip_type = 6
+                
+            backend_server_list = []
+            if 'backend' in rule_config:
+                for backend, backend_config in rule_config['backend'].items():
+                    if 'disable' in backend_config:
+                        return None
+                        
+                    if 'backend_ip' not in backend_config:
+                        raise ConfigError(err_msg + 'The backend server IP must exist!')
+                    
+                    backend_ip = backend_config['backend_ip']
+                    virtual_port = rule_config['virtual_port']
+                    backend_server = None
+                    
+                    if is_ipv4(backend_ip):
+                        backend_server = f'{backend_ip}:{virtual_port}'
```

Comment:
Please use the `bracketize_ipv6` helper https://github.com/vyos/vyos-1x/blob/current/python/vyos/template.py#L143

Reply:
done